src/neo/Network/P2P/ProtocolHandler.cs

```diff
@@ -314,6 +313,27 @@ private void OnVersionMessageReceived(VersionPayload payload)
             Context.Parent.Tell(payload);
         }
 
+        private void OnDisconnectMessageReceived(DisconnectPayload payload)
+        {
+            switch (payload.Reason)
+            {
+                case DisconnectReason.MaxConnectionReached:
+                case DisconnectReason.MaxConnectionPerAddressReached:
+                    try
+                    {
+                        var addressList = payload.Data
+                            .AsSerializableArray<NetworkAddressWithTime>(AddrPayload.MaxCountToSend)
+                            .Select(p => p.EndPoint)
+                            .Where(p => p.Port > 0);
+                        system.LocalNode.Tell(new Peer.Peers { EndPoints = addressList });
+                    }
+                    catch { }
+                    break;
+                default: break;
+            }
+            Context.Stop(Context.Parent);
```

Comment:
The child can stop its `Parent`?

Reply:
Yes, it can stop arbitrary actors in this way, which will issue a stop command to the provided actor. But it's not recommended.  
>Stopping an actor can be done by calling Context.Stop(actorRef). It is considered a bad practice to stop arbitrary actors this way. The recommended pattern is to call Context.Stop(self) inside an actor to stop itself, usually as a response to some user defined stop message or when the actor is done with its job.
https://github.com/akkadotnet/akka.net/blob/dev/docs/articles/intro/tutorial-1.md


Or should I change it to `Context.Parent.Tell(Directive.Stop, system.LocalNode);`

Reply:
I think `Context.Stop(Self);` is okay. Because when `ProtocolHandler` stopped, the `RemoteNode` will know it:

https://github.com/neo-project/neo/blob/c198dc2e7db3043a2f26967d703bc80c9a88ad56/src/neo/Network/P2P/RemoteNode.cs#L236-L243

And then the `RemoteNode` will call `Disconnect()` and stop itself.

Reply:
I will test it.  Will the `Context.Stop(Self)`  trigger the `OneForOneStrategy`? 

https://github.com/neo-project/neo/blob/c198dc2e7db3043a2f26967d703bc80c9a88ad56/src/neo/Network/P2P/RemoteNode.cs#L236-L243
It looks like, only when children actors throw exception, can trigger the strategy?

> Class OneForOneStrategy
This class represents a fault handling strategy that applies a Directive to the single child actor that failed.
https://getakka.net/api/Akka.Actor.OneForOneStrategy.html

Reply:
I'm not sure. Please test it.

Reply:
Done, the `Context.Stop(Self)` will not trigger the `SupervisorStrategy`. 

The code in parent actor
```csharp
protected override SupervisorStrategy SupervisorStrategy()
{
     return new OneForOneStrategy(ex =>
     {
           Console.WriteLine("received exception, and send Directive.Stop to child");
           return Directive.Stop;
     });
}
```
The code in child actor

```csharp
protected override void OnReceive(object message)
{
       Console.WriteLine("received msg: " + message);
       switch (message)
       {
           case Exception ex:
                throw ex;
           case Directive directive:
               Context.Stop(Self);
               break;
        }
}
```

When invoked by the following code, it will not trigger the `OneForOneStrategy`.

```csharp
Parent.Instance.Child.Tell(Directive.Stop);
```

When invoked by the following code, it will trigger the `OneForOneStrategy`

```csharp
Parent.Instance.Child.Tell(new Exception("exception"));
```

Reply:
I think the best way is to use `Context.Watch()`.

Reply:
Fixed. It worked well in local private network (4 CNs + 3 clients)

![image](https://user-images.githubusercontent.com/43396869/69939389-b6e4c380-151a-11ea-8fb3-6f7c7132b122.png)